frontend/src/metabase/query_builder/components/CompareAggregations/components/OffsetInput/OffsetInput.tsx

```diff
@@ -0,0 +1,42 @@
+import { useMemo } from "react";
+
+import { Flex, NumberInput, Text } from "metabase/ui";
+import type * as Lib from "metabase-lib";
+
+import S from "./OffsetInput.module.css";
+import { getHelp, getLabel, parsePeriodValue } from "./utils";
+
+interface Props {
+  query: Lib.Query;
+  stageIndex: number;
+  value: number | "";
+  onChange: (value: number | "") => void;
+}
+
+export const OffsetInput = ({ query, stageIndex, value, onChange }: Props) => {
+  const label = useMemo(() => getLabel(query, stageIndex), [query, stageIndex]);
+  const help = useMemo(() => getHelp(query, stageIndex), [query, stageIndex]);
+
+  return (
+    <Flex align="flex-end" pos="relative">
+      <NumberInput
+        classNames={{
+          input: S.input,
+          wrapper: S.wrapper,
+        }}
+        label={label}
+        min={1}
+        parseValue={parsePeriodValue}
```

Comment:
It doesn't seem that you need a custom `parseValue` here. Just change the value in `onChange`

Reply:
3b0c5d0895aab27d1be15ba95e89f6da832a6958